src/server/game/Warden/WardenWin.cpp

```diff
@@ -239,6 +252,80 @@ void WardenWin::HandleHashResult(ByteBuffer& buff)
     _initialized = true;
 }
 
+uint16 WardenWin::GetFreePayloadId()
+{
+    uint16 payloadId = _wardenPayloadOffset;
+
+    while (CachedChecks.find(payloadId) != CachedChecks.end())
+    {
+        payloadId++;
+    }
+
+    return payloadId;
+}
+
+/**
+* @brief Register a payload into cache and returns its payload id.
+* @param payload The payload to be stored in WardenWin::CachedChecks.
+* @return uint16 The payload id for use with Warden::QueuePayload.
+*/
+uint16 WardenWin::RegisterPayload(std::string& payload)
+{
+    uint16 payloadId = GetFreePayloadId();
+
+    WardenCheck wCheck;
+    wCheck.Type = LUA_EVAL_CHECK;
+    wCheck.Str = payload;
+    wCheck.CheckId = payloadId;
+
+    std::string idStr = Acore::StringFormat("%04u", payloadId);
+    ASSERT(idStr.size() == 4);
+    std::copy(idStr.begin(), idStr.end(), wCheck.IdStr.begin());
+
+    CachedChecks.emplace(payloadId, wCheck);
+
+    return payloadId;
+}
+
+/**
+* @brief Unregister a payload from cache and return if successful.
+* @param payloadId The payload to removed from WardenWin::CachedChecks.
+* @return bool If the payloadId was present.
+*/
+bool WardenWin::UnregisterPayload(uint16 payloadId)
+{
+    if (CachedChecks.find(payloadId) == CachedChecks.end())
```

Comment:
There's no need to check if the key exists you can just return CachedChecks.erase(payloadId);
Since it returns 0 if the value doesn't exist.

Reply:
Fixed, thank you. :)